### PR TITLE
Add comment why off-by-one error is hard to fix

### PR DESCRIPTION
**Pull request by ttencate**
See #83843.

### CODE REVIEW

File: core/io/resource_uid.cpp [hunk 1]
```diff
@@ -35,6 +35,8 @@
 #include "core/io/dir_access.h"
 #include "core/io/file_access.h"
 
+// These constants are off by 1, causing the 'z' and '9' characters never to be used.
+// This cannot be fixed without breaking compatibility; see https://github.com/godotengine/godot/issues/83843
```
Comment (akien-mga) [inline core/io/resource_uid.cpp:39]: We usually reference GitHub issues like this (and added a final dot as per out comment style).
```suggestion
// This cannot be fixed without breaking compatibility; see GH-83843.
```

**Review by akien-mga — APPROVED** [review]

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
Could you squash the commits? See [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

**Comment by ttencate** [comment]
Sorry, thanks, done. Here was me thinking I'd try out GitHub's web interface for this trivial change, and having been led to believe that you could squash upon merging.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**